<launch>

  <arg name="manager" value="cognito_nodelet_manager" />

  <arg name="method" value="load" />

  <arg name="debug" default="false" />
  <arg if="$(arg debug)" name="launch_prefix" value="xterm -e gdb --args" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />

  <node pkg="nodelet" type="nodelet" name="$(arg manager)" args="manager"
        output="screen" launch-prefix="$(arg launch_prefix)">
    <param name="num_worker_threads" value="20" />
  </node>
 
  <node pkg="usb_cam" type="usb_cam_node" name="usb_cam">
     <param name="pixel_format" value="yuyv" />
     <param name="video_device" value="/dev/video1" />
     <param name="camera_name" value="usb_cam" />
     <param name="camera_info_url" value="file://$(find mod_base)/camera.yaml" /> 
  </node> 

  <node ns="usb_cam" pkg="image_proc" type="image_proc" name="image_proc" />

  <node pkg="nodelet" type="nodelet" name="MultiObjDetector" args="$(arg method) mod_base/MultiObjDetectorNodelet $(arg manager)">
    <param name="rgb_image" value="/usb_cam/image_rect_color" />
    <param name="CodebookFilename" value="$(find mod_base)/result.eod.bin" />
    <param name="LoadCodebookFilename" value="" /> <!-- fill when needing to load -->
    <param name="Interface" value="SHOW" /> <!-- SHOW, NOSHOW -->
    <param name="PublishResults" value="PUBLISH" /> <!-- PUBLISH, NOPUBLISH -->
    <param name="WriteToFile" value="" /> <!-- a file name would result in the detections outputted to a file -->
    <param name="ImageSize" value="VGA" /> <!-- SXGA, VGA -->
    <param name="LearnObjNo" value="obj0" /> <!-- objects from 0 to 29 -->
    <param name="PathShape" value="NOSTAR" /> <!-- STAR, NOSTAR -->
  </node>

</launch>
